### ISSUE TITLE
Add support for OpenTelemetry trace context propagation in Kafka

### ISSUE THREAD

**Issue by JonasKunz** [elastic/apm-agent-java#2983]
As reported in the [discussion forums](https://discuss.elastic.co/t/apm-kafka-plugin-seems-to-only-support-binary-traceparent-header/323737), our current implementation for trace-context propagation via Kafka is incompatible with the OpenTelemetry Kafka trace context propagation.

Our implementation uses the proprietary `elasticapmtraceparent` header (see [the spec](https://github.com/elastic/apm/blob/e34a4b739dab3ee642888801ea669c501b7f3952/specs/agents/tracing-distributed-tracing.md#binary-fields)), whereas it seems like the standard opentelemetry instrumentations use the standard text-headers and encode the values in UTF8 ([java implementation example](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/10772582630ce89c66617d366d4393aedd32dfa2/instrumentation/kafka/kafka-clients/kafka-clients-common/library/src/main/java/io/opentelemetry/instrumentation/kafka/internal/KafkaHeadersSetter.java)).

We should ensure that we support the continuation of traces started in opentelemetry via kafka and also ensure that opentelemetry implementations are able to continue traces based on kafka headers injected by our java agent.

**Comment by eyalkoren** [comment]
I agree that we better support that, though it is going to be odd...
The spec should say that when we get a binary header named `tracecontext`, we need to be able to deserialize both according to the official (still draft it seems) [W3C spec](https://w3c.github.io/trace-context-binary/) as well as this form.

Either way, I think this issue belongs to the cross-agent repo - we first need to get the spec updated and then implement in each agent.
